src/Microsoft.Health.Fhir.Stu3.Core/Extensions/ModelExtensions.cs

```diff
@@ -0,0 +1,93 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Extensions
+{
+    public static class ModelExtensions
+    {
+        static ModelExtensions()
+        {
+            SetModelInfoProvider();
```

Comment:
The ambient context worries me a little bit. I think it's fine for now since it's already quite large change but we should think about this a little bit more.